hack/e2e.sh

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+ROOT=$(cd $(dirname ${BASH_SOURCE[0]})/.. && pwd)
+cd "${ROOT}"
+
+CLUSTER=${CLUSTER:-"e2e-mo-kind"}
+MO_VERSION=${MO_VERSION:-"nightly-20eeb7c9"}
+IMAGE_REPO=${IMAGE_REPO:-"matrixorigin/matrixone"}
+OPNAMESPACE=${OPNAMESPACE:-"idc-e2e-matrixone-operator"}
+TESTNAMESPACE=${TESTNAMESPACE:-"e2e"}
+
+function e2e::prepare_image() {
+    docker pull "${2}"
+    kind load docker-image --name "${1}" "${2}"
+}
+
+function e2e::kubectl_wait_appear() {
+    local WAIT_N=0
+    local MAX_WAIT=5
+    while true; do
+        kubectl get $@ 2>/dev/null | grep NAME && break
+        if [ ${WAIT_N} -lt ${MAX_WAIT} ]; then
+            WAIT_N=$((WAIT_N+1))
+            echo "Waiting for $@ to be created, sleeping for ${WAIT_N} seconds"
+            sleep ${WAIT_N}
+        else
+            echo "Timeout waiting for $@"
+            exit 1
+        fi
+    done
+}
+
+function e2e::kind-cleanup() {
+    echo "> Tearing down"
+    kind delete cluster --name ${1}
+}
+
+function e2e::idc-cleanup() {
```

Comment:
we should combine idc, kind and other kubernetes providers to a unified kubernetes abstraction